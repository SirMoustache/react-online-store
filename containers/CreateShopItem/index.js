export { default, CREATE_SHOP_ITEM_MUTATION } from './CreateShopItem';
