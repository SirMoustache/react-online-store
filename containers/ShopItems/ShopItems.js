/**
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';

export const ALL_SHOP_ITEMS_QUERY = gql`
  query ALL_SHOP_ITEMS_QUERY {
    shopItems {
      id
      title
    }
  }
`;

const ShopItemsContainer = ({ render, children }) => {
  const renderer = render || children;

  return (
    <Query query={ALL_SHOP_ITEMS_QUERY}>
      {({ error, loading, data }) => renderer({ error, loading, data })}
    </Query>
  );
};

ShopItemsContainer.propTypes = {
  render: PropTypes.func,
  children: PropTypes.func,
};

export default ShopItemsContainer;
