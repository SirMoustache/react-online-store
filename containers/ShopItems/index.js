export { default } from './ShopItems';
export { ALL_SHOP_ITEMS_QUERY } from './ShopItems';
