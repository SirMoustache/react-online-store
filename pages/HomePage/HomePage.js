/**
 * Absolute imports
 */
import React, { Fragment } from 'react';

/**
 * Containers
 */
import ShopItemsContainer from '../../containers/ShopItems';

/**
 * Components
 */
import ShopItemOverview from '../../components/ShopItemOverview';

const HomePage = () => (
  <Fragment>
    <h1>Hello World from Home Page</h1>
    <ShopItemsContainer
      render={data => {
        console.log('Apollo data: ', data);
        return <ShopItemOverview />;
      }}
    />
  </Fragment>
);

export default HomePage;
