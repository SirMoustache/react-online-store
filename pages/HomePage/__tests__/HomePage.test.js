/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MockedProvider } from 'react-apollo/test-utils';

/**
 * Local Components
 */
import HomePage from '..';

/**
 * GraphQL
 */
import { ALL_SHOP_ITEMS_QUERY } from '../../../containers/ShopItems';

const mocks = [
  {
    request: { query: ALL_SHOP_ITEMS_QUERY },
    result: {
      data: {},
    },
  },
];

const renderComponent = ({ ...rest } = {}) =>
  render(
    <MockedProvider mocks={mocks}>
      <HomePage {...rest} />
    </MockedProvider>,
  );

describe('<HomePage />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
