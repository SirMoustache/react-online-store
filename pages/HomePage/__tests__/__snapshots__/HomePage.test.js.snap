// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match snapshot 1`] = `
<div>
  <h1>
    Hello World from Home Page
  </h1>
  <span>
    sp
  </span>
</div>
`;
